zerver/webhooks/github/view.py

```diff
@@ -395,46 +406,50 @@ def get_status_body(helper: Helper) -> str:
     payload = helper.payload
     if payload["target_url"]:
         status = "[{}]({})".format(
-            payload["state"],
-            payload["target_url"],
+            payload["state"].tame(check_string),
+            payload["target_url"].tame(check_string),
         )
     else:
-        status = payload["state"]
+        status = payload["state"].tame(check_string)
     return "[{}]({}) changed its status to {}.".format(
-        payload["sha"][:7],  # TODO
-        payload["commit"]["html_url"],
+        payload["sha"].tame(check_string)[:7],  # TODO
+        payload["commit"]["html_url"].tame(check_string),
         status,
     )
 
 
 def get_locked_or_unlocked_pull_request_body(helper: Helper) -> str:
     payload = helper.payload
 
-    action = payload["action"]
+    action = payload["action"].tame(check_string)
 
     message = "{sender} has locked [PR #{pr_number}]({pr_url}) as {reason} and limited conversation to collaborators."
     if action == "unlocked":
         message = "{sender} has unlocked [PR #{pr_number}]({pr_url})."
+    if payload["pull_request"]["active_lock_reason"]:
+        active_lock_reason = payload["pull_request"]["active_lock_reason"].tame(check_string)
+    else:
+        active_lock_reason = None
```

Comment:
This pattern incorrectly converts `""` to `None`. The correct way to tame a value that’s `None` or `str` is `.tame(check_none_or(check_string))`.